b-isdn-feed-config-opkg: drop package in favor of Yocto vars

KanjiMonster opened the pull request:
Instead of providing the opkg feed configuration via a custom package, Yocto also allows setting up feed configuration via configuration variables.

So let's do that instead to reduce complexity, and just reuse existing facilities.

This slightly changes the configuration, as the feeds will now be named `uri-<arch>-0`.

cat /etc/opkg/base-feeds.conf

before:

```
src/gz 5.3.1-all http://repo.bisdn.de/latest/generic-armel-iproc/packages_latest-build/ipk/all
src/gz 5.3.1-cortexa9-vfp http://repo.bisdn.de/latest/generic-armel-iproc/packages_latest-build/ipk/cortexa9-vfp
src/gz 5.3.1-generic_armel_iproc http://repo.bisdn.de/latest/generic-armel-iproc/packages_latest-build/ipk/generic_armel_iproc
```

after:

```
src/gz uri-all-0 http://repo.bisdn.de/latest/generic-armel-iproc/packages_latest-build/ipk/all
src/gz uri-cortexa9-vfp-0 http://repo.bisdn.de/latest/generic-armel-iproc/packages_latest-build/ipk/cortexa9-vfp
src/gz uri-generic_armel_iproc-0 http://repo.bisdn.de/latest/generic-armel-iproc/packages_latest-build/ipk/generic_armel_iproc
```

jklare commented:
For me, it looks like the feeds before were suggesting a version number ("5.3.1" in your example) to the user, while the new feeds seem to omit that and only have "uri-" instead. Since the source of both feeds are pointing to the latest anyway, it might be that this information before was rather misleading, but i am not sure if there is some other meta information somewhere that I don't see.

KanjiMonster commented:
That's more or less everything there is to it. Before, we used `FEEDNAME_PREFIX` (which was always set to `${DISTRO_VERSION}`), now it's set by Yocto to `uri`.

A release base-feeds.conf would look like this:

```
src/gz uri-all-0 http://repo.bisdn.de/pub/onie/generic-armel-iproc/packages-v5.3.1/ipk/all
src/gz uri-cortexa9-vfp-0 http://repo.bisdn.de/pub/onie/generic-armel-iproc/packages-v5.3.1/ipk/cortexa9-vfp
src/gz uri-generic_armel_iproc-0 http://repo.bisdn.de/pub/onie/generic-armel-iproc/packages-v5.3.1/ipk/generic_armel_iproc
```

`opkg` prints both parts (name and uri) when fetching the feeds:

```
accton-as4610-54:~$ sudo opkg update
Downloading http://repo.bisdn.de/nightly_builds/generic-armel-iproc/main/packages_latest-build/ipk/all/Packages.gz.
Updated source 'uri-all-0'.
Downloading http://repo.bisdn.de/nightly_builds/generic-armel-iproc/main/packages_latest-build/ipk/cortexa9-vfp/Packages.gz.
Updated source 'uri-cortexa9-vfp-0'.
Downloading http://repo.bisdn.de/nightly_builds/generic-armel-iproc/main/packages_latest-build/ipk/generic_armel_iproc/Packages.gz.
Updated source 'uri-generic_armel_iproc-0'.
```

with the URI more prominently, so IMHO it's still very obvious where the feeds point to (and if release feeds, which release it is).
